python/ray/tests/test_dynres.py

```diff
@@ -569,3 +569,25 @@ def delete_res(resource_name, res_client_id):
             all_resources_created.append(str(i) in resources)
         success = all(all_resources_created)
     assert success
+
+
+def test_release_cpus_when_actor_creation_task_blocking(shutdown_only):
+    ray.init(num_cpus=2)
+
+    @ray.remote(num_cpus=1)
+    def get_100():
+        import time
+        time.sleep(2)
+        return 100
+
+    @ray.remote(num_cpus=1)
+    class A(object):
+        def __init__(self):
+            self.num = ray.get(get_100.remote())
+
+        def get_num(self):
+            return self.num
+
+    a = A.remote()
+    assert 100 == ray.get(a.get_num.remote())
+    assert 1 == ray.available_resources()["CPU"]
```

Comment:
This test will be flaky because resource information is broadcast only once every hundred milliseconds or so. I'd suggest looping until the assert passes or some longer timeout expires.

Reply:
Nice.